aii-pxelinux/src/main/perl/PXELINUX/constants.pm

```diff
@@ -0,0 +1,80 @@
+#${PMpre} NCM::Component::PXELINUX::constants${PMpost}
+
+=pod
+
+This aii-pxelinux submodule only defines constants used by
+pxelinux module and shared with others. It allows to share
+the constant definitions without bringing all the aii-pxelinux
+depencies.
+
+=cut
+
+use Readonly;
+use Exporter qw(import);
+
+# PXE commands supported
+use constant CONFIGURE => undef;
+use constant INSTALL => 'install';
+use constant BOOT => 'boot';
+use constant RESCUE => 'rescue';
+use constant RESCUEBOOT => 'rescueconfig';
+use constant FIRMWARE => 'firmware';
+use constant LIVECD => 'livecd';
+
+# Default paths and option values related to PXE configuration
+use constant OSINSTALL_DEF_ROOT_PATH => '/osinstall/nbp';
+use constant OSINSTALL_DEF_PXELINUX_DIR => '/pxelinux.cfg';
+use constant OSINSTALL_DEF_GRUB2_DIR => '/grub-efi';
+use constant LOCAL_BOOT_CONFIG_FILE => 'localboot.cfg';
+use constant GRUB2_EFI_LINUX_CMD_DEFAULT => 'linuxefi';
+use constant NBPDIR_VARIANT_DISABLED => 'none';
+
+# Option names related to PXE configuration
+use constant LOCALBOOT => 'bootconfig';
+use constant NBPDIR_PXELINUX => 'nbpdir';
+use constant NBPDIR_GRUB2 => 'nbpdir_grub2';
+use constant GRUB2_EFI_KERNEL_ROOT => 'grub2_efi_kernel_root';
+use constant GRUB2_EFI_LINUX_CMD => 'grub2_efi_linux_cmd';
+use constant GRUB2_EFI_INITRD_CMD => 'grub2_efi_initrd_cmd';
+
+# Supported PXE variants 
+use enum qw(PXE_VARIANT_PXELINUX PXE_VARIANT_GRUB2);
+
+# To ease constant export/import by tests and other modules
+# Some constants are use only by unit tests
+Readonly my @PXE_VARIANTS => qw(
```

Comment:
i'm not sure why (maybe old EL5 ism), but we use `Readonly::Array` for arrays and `Readonly::Hash` elsewhere.
if it was a EL5-ism, this is ok now since aii is now el6+